src/layout.c

```diff
@@ -418,9 +455,16 @@ void layout(void) {
 			break;
 
 		case CANCELLED:
-			log_warn("\nChanges cancelled, retrying");
 			displ->state = IDLE;
-			return;
+
+			retries++;
+			if (retries <= MAX_CANCEL_RETRIES) {
+				log_warn("\nChanges cancelled, retrying (attempt %i)", retries);
+				break;
+			} else {
+				log_warn("\nChanges cancelled, max number of retry attempts exceeded");
+				return;
+			}
```

Comment:
Can we please move this change to a separate PR?

We can release this or the conditions separately, as well as revert them.